features/util.py

```diff
@@ -161,3 +163,57 @@ def lxc_get_series(name: str, image: bool = False):
                 " Could not detect image series. Add it via `lxc image edit`"
             )
     return None
+
+
+def lxc_build_deb(container_name: str, output_deb_file: str) -> None:
+    """
+    Push source PR code .tar.gz to the container.
+    Run tools/build-from-source.sh which will create the .deb
+    Pull .deb from this container to travis-ci instance
+
+    :param container_name: the name of the container to:
+         - push the PR source code;
+         - pull the built .deb package.
+    :param output_deb_file: the new output .deb from source code
+    """
+
+    print("\n\n\n LXC file push {}".format(SOURCE_PR_TGZ))
+    subprocess.run(
+        ["lxc", "file", "push", SOURCE_PR_TGZ, container_name + "/tmp/"]
+    )
+    script = "build-from-source.sh"
+    with open(script, "w") as stream:
+        stream.write(
+            textwrap.dedent(
+                """\
+            #!/bin/bash
+            set -o xtrace
+            apt-get update
+            apt-get install make
+            cd /tmp
+            tar -zxvf *gz
+            cd ubuntu-advantage-client
+            make deps
+            dpkg-buildpackage -us -uc
+            cp /tmp/ubuntu-advantage-tools*.deb /tmp/ubuntu-advantage-tools.deb
```

Comment:
Maybe I am missing something here, but if the `dpkg-buildpackage -us -uc` produces more than one deb files, like:

* ubuntu-advantage-pro_25.0_amd64.deb
* ubuntu-advantage-tools_25.0_amd64.deb

Would the cp command work in that situation ?

Reply:
Yes. It does, but for PR, as we are running only on travis-CI we do not need the -pro.deb.
So, we need to install the -tools.deb.

Reply:
I completely missed that the pro image does not have the `tools` part in the name :facepalm: 
Sorry @patriciadomin

Reply:
no worries @lucasmoura